.github/workflows/rust.yml

```diff
@@ -0,0 +1,28 @@
+name: Rust
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup
+      run: |
+              sudo apt-get update -q
+              sudo apt-get -y install secilc
+    - name: Build
+      run: cargo build --verbose
+    - name: Run tests
+      run: cargo test --verbose
+    - name: Check format
+      run: cargo fmt --all -- --check
```

Comment:
Other steps could be inspired from https://github.com/dburgener/cascade/blob/main/.pipelines/OneBranch.PullRequest.yml#L48-L85